File: src/types/comunicazione.ts
// Types per il sistema di comunicazione

// ============================================================================
// TEMPLATE MESSAGGI
// ============================================================================

export interface TemplateMesaggio {
  id: string;
  codice: string;
  nome: string;
  tipo: 'reminder' | 'birthday' | 'marketing' | 'custom';
  canale: 'whatsapp' | 'email';
  oggetto: string | null;
  corpo: string;
  attivo: boolean;
  created_at: string;
  updated_at: string;
}

export interface CreateTemplateInput {
  codice: string;
  nome: string;
  tipo: string;
  canale: string;
  oggetto?: string;
  corpo: string;
}

export interface UpdateTemplateInput {
  nome?: string;
  oggetto?: string;
  corpo?: string;
  attivo?: boolean;
}

// ============================================================================
// COMUNICAZIONE (Log messaggi)
// ============================================================================

export interface Comunicazione {
  id: string;
  cliente_id: string;
  appuntamento_id: string | null;
  template_id: string | null;
  tipo: string;
  canale: string;
  stato: 'in_coda' | 'inviato' | 'consegnato' | 'letto' | 'errore';
  destinatario: string;
  oggetto: string | null;
  messaggio: string;
  inviato_at: string | null;
  errore_messaggio: string | null;
  created_at: string;
}

export interface ComunicazioneWithCliente extends Comunicazione {
  cliente_nome: string;
  cliente_cognome: string;
}

// ============================================================================
// CONFIG SMTP
// ============================================================================

export interface ConfigSmtp {
  id: string;
  host: string;
  port: number;
  username: string;
  password: string;
  from_email: string;
  from_name: string | null;
  encryption: 'tls' | 'ssl';
  enabled: boolean;
  created_at: string;
  updated_at: string;
}

export interface SaveSmtpConfigInput {
  host: string;
  port: number;
  username: string;
  password: string;
  from_email: string;
  from_name?: string;
  encryption: string;
  enabled: boolean;
}

// ============================================================================
// CONFIG SCHEDULER
// ============================================================================

export interface ConfigScheduler {
  id: string;
  reminder_enabled: boolean;
  reminder_hours_before: number;
  reminder_second_hours_before: number | null;
  reminder_default_channel: string;
  birthday_enabled: boolean;
  birthday_check_time: string;
  birthday_default_channel: string;
  birthday_template_id: string | null;
  last_reminder_check: string | null;
  last_birthday_check: string | null;
  created_at: string;
  updated_at: string;
}

export interface SaveSchedulerConfigInput {
  reminder_enabled: boolean;
  reminder_hours_before: number;
  reminder_second_hours_before?: number;
  reminder_default_channel: string;
  birthday_enabled: boolean;
  birthday_check_time: string;
  birthday_default_channel: string;
  birthday_template_id?: string;
}

// ============================================================================
// CAMPAGNE MARKETING
// ============================================================================

export interface CampagnaMarketing {
  id: string;
  nome: string;
  descrizione: string | null;
  target_filters: string | null;
  template_id: string | null;
  canale: string;
  messaggio_personalizzato: string | null;
  oggetto_email: string | null;
  tipo_invio: 'immediato' | 'programmato';
  data_invio_programmato: string | null;
  stato: 'bozza' | 'in_corso' | 'completata' | 'annullata';
  totale_destinatari: number;
  inviati: number;
  consegnati: number;
  aperti: number;
  errori: number;
  creato_da: string | null;
  avviata_at: string | null;
  completata_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface CreateCampagnaInput {
  nome: string;
  descrizione?: string;
  canale: string;
  template_id?: string;
  messaggio_personalizzato?: string;
  oggetto_email?: string;
  target_filters?: string;
  tipo_invio?: string;
  data_invio_programmato?: string;
}

export interface CampagnaDestinatario {
  id: string;
  campagna_id: string;
  cliente_id: string;
  comunicazione_id: string | null;
  stato: string;
  errore_messaggio: string | null;
  inviato_at: string | null;
  created_at: string;
}

// ============================================================================
// MESSAGE LINK
// ============================================================================

export interface MessageLink {
  link: string;
  canale: string;
  destinatario: string;
  messaggio: string;
}

// ============================================================================
// STATISTICHE
// ============================================================================

export interface ComunicazioniStats {
  totale: number;
  inviati: number;
  consegnati: number;
  errori: number;
  oggi: number;
  questa_settimana: number;
  per_tipo: TipoCount[];
  per_canale: CanaleCount[];
}

export interface TipoCount {
  tipo: string;
  count: number;
}

export interface CanaleCount {
  canale: string;
  count: number;
}

// ============================================================================
// FILTRI
// ============================================================================

export interface FiltriComunicazioni {
  cliente_id?: string;
  tipo?: string;
  canale?: string;
  stato?: string;
  data_da?: string;
  data_a?: string;
}

export interface TargetFilters {
  con_consenso_marketing?: boolean;
  con_appuntamenti_recenti?: boolean;
  giorni_ultima_visita_min?: number;
  giorni_ultima_visita_max?: number;
  trattamenti_ids?: string[];
  categorie_trattamenti_ids?: string[];
}

// ============================================================================
// HELPERS
// ============================================================================

export const CANALI = [
  { value: 'whatsapp', label: 'WhatsApp', icon: 'MessageCircle' },
  { value: 'email', label: 'Email', icon: 'Mail' },
] as const;

export const TIPI_TEMPLATE = [
  { value: 'reminder', label: 'Promemoria Appuntamento' },
  { value: 'birthday', label: 'Auguri Compleanno' },
  { value: 'marketing', label: 'Marketing' },
  { value: 'custom', label: 'Personalizzato' },
] as const;

export const STATI_COMUNICAZIONE = [
  { value: 'pending', label: 'In attesa', color: 'yellow' },
  { value: 'inviato', label: 'Inviato', color: 'blue' },
  { value: 'consegnato', label: 'Consegnato', color: 'green' },
  { value: 'errore', label: 'Errore', color: 'red' },
] as const;

export const STATI_CAMPAGNA = [
  { value: 'bozza', label: 'Bozza', color: 'gray' },
  { value: 'in_corso', label: 'In corso', color: 'blue' },
  { value: 'completata', label: 'Completata', color: 'green' },
  { value: 'annullata', label: 'Annullata', color: 'red' },
] as const;

// Placeholder disponibili nei template
export const TEMPLATE_PLACEHOLDERS = [
  { placeholder: '{nome}', descrizione: 'Nome del cliente' },
  { placeholder: '{cognome}', descrizione: 'Cognome del cliente' },
  { placeholder: '{data_appuntamento}', descrizione: 'Data appuntamento (dd/mm/yyyy)' },
  { placeholder: '{ora_appuntamento}', descrizione: 'Ora appuntamento (HH:MM)' },
  { placeholder: '{trattamento}', descrizione: 'Nome del trattamento' },
  { placeholder: '{nome_centro}', descrizione: 'Nome del centro estetico' },
] as const;

File: src-tauri/src/models/comunicazione.rs
// Modelli per il sistema di comunicazione
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

// ============================================================================
// TEMPLATE MESSAGGI
// ============================================================================

#[derive(Debug, Serialize, Deserialize, FromRow, Clone)]
pub struct TemplateMesaggio {
    pub id: String,
    pub codice: String,
    pub nome: String,
    pub tipo: String,
    pub canale: String,
    pub oggetto: Option<String>,
    pub corpo: String,
    pub attivo: bool,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct CreateTemplateInput {
    pub codice: String,
    pub nome: String,
    pub tipo: String,
    pub canale: String,
    pub oggetto: Option<String>,
    pub corpo: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct UpdateTemplateInput {
    pub nome: Option<String>,
    pub oggetto: Option<String>,
    pub corpo: Option<String>,
    pub attivo: Option<bool>,
}

// ============================================================================
// COMUNICAZIONE (Log messaggi inviati)
// ============================================================================

#[derive(Debug, Serialize, Deserialize, FromRow, Clone)]
pub struct Comunicazione {
    pub id: String,
    pub cliente_id: String,
    pub appuntamento_id: Option<String>,
    pub campagna_id: Option<String>,
    pub template_id: Option<String>,
    pub tipo: String,
    pub canale: String,
    pub stato: String,
    pub destinatario: String,
    pub oggetto: Option<String>,
    pub messaggio: String,
    pub inviato_at: Option<DateTime<Utc>>,
    pub consegnato_at: Option<DateTime<Utc>>,
    pub letto_at: Option<DateTime<Utc>>,
    pub errore_messaggio: Option<String>,
    pub provider_id: Option<String>,
    pub provider_response: Option<String>,
    pub costo: Option<f64>,
    pub created_at: DateTime<Utc>,
}

#[derive(Debug, Serialize, Deserialize, FromRow, Clone)]
pub struct ComunicazioneWithCliente {
    pub id: String,
    pub cliente_id: String,
    pub cliente_nome: String,
    pub cliente_cognome: String,
    pub appuntamento_id: Option<String>,
    pub template_id: Option<String>,
    pub tipo: String,
    pub canale: String,
    pub stato: String,
    pub destinatario: String,
    pub oggetto: Option<String>,
    pub messaggio: String,
    pub inviato_at: Option<DateTime<Utc>>,
    pub errore_messaggio: Option<String>,
    pub created_at: DateTime<Utc>,
}

// ============================================================================
// CONFIG SMTP
// ============================================================================

#[derive(Debug, Serialize, Deserialize, FromRow, Clone)]
pub struct ConfigSmtp {
    pub id: String,
    pub host: String,
    pub port: i32,
    pub username: String,
    pub password: String,
    pub from_email: String,
    pub from_name: Option<String>,
    pub encryption: String,
    pub enabled: bool,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct SaveSmtpConfigInput {
    pub host: String,
    pub port: i32,
    pub username: String,
    pub password: String,
    pub from_email: String,
    pub from_name: Option<String>,
    pub encryption: String,
    pub enabled: bool,
}

// ============================================================================
// CONFIG SCHEDULER
// ============================================================================

#[derive(Debug, Serialize, Deserialize, FromRow, Clone)]
pub struct ConfigScheduler {
    pub id: String,
    pub reminder_enabled: bool,
    pub reminder_hours_before: i32,
    pub reminder_second_hours_before: Option<i32>,
    pub reminder_default_channel: String,
    pub birthday_enabled: bool,
    pub birthday_check_time: String,
    pub birthday_default_channel: String,
    pub birthday_template_id: Option<String>,
    pub last_reminder_check: Option<DateTime<Utc>>,
    pub last_birthday_check: Option<DateTime<Utc>>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct SaveSchedulerConfigInput {
    pub reminder_enabled: bool,
    pub reminder_hours_before: i32,
    pub reminder_second_hours_before: Option<i32>,
    pub reminder_default_channel: String,
    pub birthday_enabled: bool,
    pub birthday_check_time: String,
    pub birthday_default_channel: String,
    pub birthday_template_id: Option<String>,
}

// ============================================================================
// CAMPAGNE MARKETING
// ============================================================================

#[derive(Debug, Serialize, Deserialize, FromRow, Clone)]
pub struct CampagnaMarketing {
    pub id: String,
    pub nome: String,
    pub descrizione: Option<String>,
    pub target_filters: Option<String>,
    pub template_id: Option<String>,
    pub canale: String,
    pub messaggio_personalizzato: Option<String>,
    pub oggetto_email: Option<String>,
    pub tipo_invio: String,
    pub data_invio_programmato: Option<DateTime<Utc>>,
    pub stato: String,
    pub totale_destinatari: i32,
    pub inviati: i32,
    pub consegnati: i32,
    pub aperti: i32,
    pub errori: i32,
    pub creato_da: Option<String>,
    pub avviata_at: Option<DateTime<Utc>>,
    pub completata_at: Option<DateTime<Utc>>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct CreateCampagnaInput {
    pub nome: String,
    pub descrizione: Option<String>,
    pub canale: String,
    pub template_id: Option<String>,
    pub messaggio_personalizzato: Option<String>,
    pub oggetto_email: Option<String>,
    pub target_filters: Option<String>,
    pub tipo_invio: Option<String>,
    pub data_invio_programmato: Option<String>,
}

#[derive(Debug, Serialize, Deserialize, FromRow, Clone)]
pub struct CampagnaDestinatario {
    pub id: String,
    pub campagna_id: String,
    pub cliente_id: String,
    pub comunicazione_id: Option<String>,
    pub stato: String,
    pub errore_messaggio: Option<String>,
    pub inviato_at: Option<DateTime<Utc>>,
    pub created_at: DateTime<Utc>,
}

// ============================================================================
// MESSAGE LINK (Output per generatori link)
// ============================================================================

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct MessageLink {
    pub link: String,
    pub canale: String,
    pub destinatario: String,
    pub messaggio: String,
}

// ============================================================================
// STATISTICHE
// ============================================================================

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct ComunicazioniStats {
    pub totale: i32,
    pub inviati: i32,
    pub consegnati: i32,
    pub errori: i32,
    pub oggi: i32,
    pub questa_settimana: i32,
    pub per_tipo: Vec<TipoCount>,
    pub per_canale: Vec<CanaleCount>,
}

#[derive(Debug, Serialize, Deserialize, Clone, FromRow)]
pub struct TipoCount {
    pub tipo: String,
    pub count: i32,
}

#[derive(Debug, Serialize, Deserialize, Clone, FromRow)]
pub struct CanaleCount {
    pub canale: String,
    pub count: i32,
}

// ============================================================================
// FILTRI
// ============================================================================

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct FiltriComunicazioni {
    pub cliente_id: Option<String>,
    pub tipo: Option<String>,
    pub canale: Option<String>,
    pub stato: Option<String>,
    pub data_da: Option<String>,
    pub data_a: Option<String>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct TargetFilters {
    pub con_consenso_marketing: Option<bool>,
    pub con_appuntamenti_recenti: Option<bool>,
    pub giorni_ultima_visita_min: Option<i32>,
    pub giorni_ultima_visita_max: Option<i32>,
    pub trattamenti_ids: Option<Vec<String>>,
    pub categorie_trattamenti_ids: Option<Vec<String>>,
}
